### PR TITLE
Add user authorization with cancan gem

### PR DESCRIPTION
**Pull request by Berkay-akbas**
# Hello Code Reviewer 👋 

# Changes ♻️ 
- New column "role" added to users table with migration.
- If user is admin they can manage all posts and comments.
- Users now can delete their own posts and comments. Delete button won't be visible if they are not the author.

Thanks for reviewing our code!

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def initialize(user)
+    return unless user.present?
+
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, author_id: user.id
+    can :manage, Like, author_id: user.id
+    can :manage, User, id: user.id
+    can :manage, :all if user.role == 'admin'
+  end
+end
```
Comment (OLIPLICHE) [inline app/models/ability.rb:13]: - Please I need to test your code, Unfortunately, there is an error coming up when trying to Sign up. Kindly fix. 
- 
Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Berkay-akbas, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used 👏


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (Berkay-akbas) [reply app/models/ability.rb:13]: @OLIPLICHE Thanks for the review. I was unsuccessful to recreate the error you are getting. Everything seems to work on my end. I'll just request another review and see what they will get.

Comment (codepantha) [inline app/models/ability.rb:13]: - [ ] Just like the previous reviewer highlighted, there's an error 🐛 when creating a new account. It appears that Devise is protesting a redirect. As a result, I am unable to fully test your code. Kindly see the error below 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @Berkay-akbas 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Used GitFlow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (Berkay-akbas) [reply app/models/ability.rb:13]: @codepantha Thank you very much. Could you please try again? I think I found the culprit but I'm still unable to test it. It still works on my end.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @Berkay-akbas and @developerwaleed 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**